python/.pre-commit-config.yaml

```diff
@@ -0,0 +1,47 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+---
+files: ^python/
```

Comment:
Do we need this? For example I use `python/` as my working directory and when I run the `make lint` command it causes it to skip everything. I understand the CI environment may need these paths so its fine to have it in  `.github/workflows/python-ci.yml` but is there a way to avoid this in here?

Reply:
Hey @dhruv-pratap. Yes, we need this, otherwise, it will touch most of the repo (including all the Java files) because of trailing whitespaces and end-of-line checks. We could remove this, but then it should be a separate PR.

It is skipping everything on your end because it will only check the files you've touched. Pre-commit does this based on git (it checks like `git status`). This is helpful when the project becomes bigger, and the checks take more time. Otherwise, we can also disable this by always running `pre-commit run --all-files`. Let me know what you think!

Reply:
My bad, I didn't realize that. I think we can leave it like it is in that case. Thanks!